File: BackEnd/backend/src/main/java/com/fitapp/backend/sport/aplication/dto/request/SportFilterRequest.java
package com.fitapp.backend.sport.aplication.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fitapp.backend.infrastructure.persistence.entity.enums.SportSourceType;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import org.springframework.data.domain.Sort;

import java.util.ArrayList;
import java.util.List;

@Data
@Schema(description = "Filtros para búsqueda de deportes")
public class SportFilterRequest {
    
    @Schema(description = "Texto de búsqueda en nombre", example = "fut")
    @JsonProperty("search")
    private String search;
    
    @Schema(description = "Si es predefinido o no")
    @JsonProperty("isPredefined")
    private Boolean isPredefined;
    
    @Schema(description = "Tipo de origen", example = "OFFICIAL")
    @JsonProperty("sourceType")
    private SportSourceType sourceType;
    
    @Schema(description = "Creado por usuario específico")
    @JsonProperty("createdBy")
    private Long createdBy;
    
    @Schema(description = "Página solicitada (0-index)", example = "0")
    @JsonProperty("page")
    private Integer page = 0;
    
    @Schema(description = "Tamaño de página", example = "10")
    @JsonProperty("size")
    private Integer size = 10;
    
    @Schema(description = "Campo para ordenar", example = "name")
    @JsonProperty("sortBy")
    private String sortBy = "name";
    
    @Schema(description = "Dirección del orden", example = "ASC")
    @JsonProperty("direction")
    private Sort.Direction direction = Sort.Direction.ASC;
    
    @Schema(description = "Múltiples campos para ordenar")
    @JsonProperty("sortFields")
    private List<SortField> sortFields = new ArrayList<>();
    
    @Data
    public static class SortField {
        @JsonProperty("field")
        private String field;
        
        @JsonProperty("direction")
        private Sort.Direction direction = Sort.Direction.ASC;
    }
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/sport/request/SportFilterRequest.kt
package com.fitapp.appfit.response.sport.request

import com.google.gson.annotations.SerializedName

data class SportFilterRequest(
    @SerializedName("search") val search: String? = null,
    @SerializedName("category") val category: String? = null,
    @SerializedName("isPredefined") val isPredefined: Boolean? = null,
    @SerializedName("sourceType") val sourceType: String? = null,
    @SerializedName("createdBy") val createdBy: Long? = null,
    @SerializedName("page") val page: Int = 0,
    @SerializedName("size") val size: Int = 10,
    @SerializedName("sortBy") val sortBy: String = "name",
    @SerializedName("direction") val direction: String = "ASC"
)